src/SelectPicker/test/SelectPickerSpec.js

```diff
@@ -400,4 +404,53 @@ describe('SelectPicker', () => {
       expect(getByTestId('content')).to.have.text('Not selected');
     });
   });
+  it('SearchWord should be reset when controlled and triggered off', done => {
+    let searchRef = '';
+    let onClose = null;
+    const promise = new Promise(resolve => {
+      onClose = resolve;
+    });
+    const Wrapper = () => {
+      const [search, setSearch] = useState(searchRef);
+      const containerRef = useRef();
+      searchRef = search;
+      const handleSearch = value => {
+        setSearch(value);
+      };
+      const handleClose = () => {
+        onClose();
+      };
+      const container = () => containerRef.current;
+      return (
+        <div ref={containerRef}>
+          <button id="exit">exit</button>
+          <SelectPicker
+            container={container}
+            search={search}
+            defaultOpen
+            onClose={handleClose}
+            onSearch={handleSearch}
+            data={data}
+          />
+        </div>
+      );
+    };
+    Wrapper.displayName = 'WrapperSelectPicker';
+    const { container } = render(<Wrapper />);
+
+    const exit = container?.querySelector('#exit');
+    const input = container.querySelector('.rs-picker-search-bar-input');
+
+    // change search
+    fireEvent.change(input, { target: { value: 'a' } });
+    assert.equal(searchRef, 'a');
+
+    // close select
+    fireEvent.mouseDown(exit, { bubbles: true });
+
+    promise.then(() => {
+      assert.equal(searchRef, '');
+      done();
+    });
+  });
```

Comment:
Please refer to existing test scripts and see how to use `sinon.spy()` to validate a callback has been called with correct arguments.

Reply:
I modified it, but I think this test code does not reflect its operation scenario, How do you think

Reply:
I mean the way this is written should not be suitable for such a description, should is "shoule call onSearch when closed", or other

Reply:
> I mean the way this is written should not be suitable for such a description, should is "shoule call onSearch when closed", or other

The description was not quite suitable in the first place. "Should call onSearch with '' when closed" is exactly what should be tested here. The goal of unit tests is to make sure the APIs we provide (`onSearch`) work as expected (called with '' when closed), rather than to cover how users are using our APIs (e.g. updating a state when onSearch is called).